src/main/scala/za/co/absa/commons/spark/SchemaUtils.scala

```diff
@@ -0,0 +1,214 @@
+  private def diffField(field1: StructField, field2: StructField, parent: String): Seq[String] = {
+    field1.dataType match {
+      case _ if field1.dataType.typeName != field2.dataType.typeName =>
+        Seq(s"$parent.${field1.name} data type doesn't match (${field1.dataType.typeName}) vs (${field2.dataType.typeName})")
+      case arrayType1: ArrayType =>
+        diffArray(arrayType1, field2.dataType.asInstanceOf[ArrayType], s"$parent.${field1.name}")
+      case structType1: StructType =>
+        diffSchema(structType1, field2.dataType.asInstanceOf[StructType], s"$parent.${field1.name}")
+      case _ =>
+        Seq.empty[String]
+    }
+  }
+
+  /**
+   * Returns data selector that can be used to align schema of a data frame. You can use [[alignSchema]].
+   *
+   * @param schema Schema that serves as the model of column order
+   * @return Sorted DF to conform to schema
+   */
+  def getDataFrameSelector(schema: StructType): List[Column] = {
+    import za.co.absa.spark.hofs._
+
+    def processArray(arrType: ArrayType, column: Column, name: String): Column = {
+      arrType.elementType match {
+        case arrType: ArrayType =>
+          transform(column, x => processArray(arrType, x, name)).as(name)
+        case nestedStructType: StructType =>
+          transform(column, x => struct(processStruct(nestedStructType, Some(x)): _*)).as(name)
+        case _ => column
+      }
+    }
+
+    def processStruct(curSchema: StructType, parent: Option[Column]): List[Column] = {
+      curSchema.foldRight(List.empty[Column])((field, acc) => {
+        val currentCol: Column = parent match {
+          case Some(x) => x.getField(field.name).as(field.name)
+          case None    => col(field.name)
+        }
+        field.dataType match {
+          case arrType: ArrayType     => processArray(arrType, currentCol, field.name) :: acc
+          case structType: StructType => struct(processStruct(structType, Some(currentCol)): _*).as(field.name) :: acc
+          case _                      =>  currentCol :: acc
+        }
+      })
+    }
+
+    processStruct(schema, None)
+  }
+
+  /**
+   * Using schema selector from [[getDataFrameSelector]] aligns the schema of a DataFrame to the selector for operations
+   * where schema order might be important (e.g. hashing the whole rows and using except)
+   *
+   * @param df DataFrame to have it's schema aligned/sorted
+   * @param selector Selector for the DataFrame
+   * @return Returns aligned and filtered schema
+   */
+  def alignSchema(df: DataFrame, selector: List[Column]): DataFrame = df.select(selector: _*)
+
+  /**
+   * Compares 2 dataframe schemas.
+   *
+   * @param schema1 The first schema to compare
+   * @param schema2 The second schema to compare
+   * @return true if provided schemas are the same ignoring nullability
+   */
+  def isSameSchema(schema1: StructType, schema2: StructType): Boolean = {
+    val fields1 = schema1.map(field => field.name.toLowerCase() -> field).toMap
```

Comment:
In some cases DataFrame can have more columns with the same name. This would probably ignore some of those columns. [Example](https://stackoverflow.com/questions/33778664/spark-dataframe-distinguish-columns-with-duplicated-name)